app/src/main/java/fr/neamar/kiss/CustomIconDialog.java

```diff
@@ -0,0 +1,455 @@
+        mIconGrid.setVisibility(mIconData.isEmpty() ? View.GONE : View.VISIBLE);
+        ((BaseAdapter) mIconGrid.getAdapter()).notifyDataSetChanged();
+    }
+
+    static class IconData {
+        final IconPackXML.DrawableInfo drawableInfo;
+        final IconPackXML iconPack;
+
+        IconData(IconPackXML iconPack, IconPackXML.DrawableInfo drawableInfo) {
+            this.iconPack = iconPack;
+            this.drawableInfo = drawableInfo;
+        }
+
+        Drawable getIcon() {
+            return iconPack.getDrawable(drawableInfo);
+        }
+    }
+
+    static class IconAdapter extends BaseAdapter {
+        private final List<IconData> mIcons;
+        private OnItemClickListener mOnItemClickListener = null;
+
+        public interface OnItemClickListener {
+            void onItemClick(IconAdapter adapter, View view, int position);
+        }
+
+        IconAdapter(@NonNull List<IconData> objects) {
+            mIcons = objects;
+        }
+
+        void setOnItemClickListener(OnItemClickListener listener) {
+            mOnItemClickListener = listener;
+        }
+
+        @Override
+        public IconData getItem(int position) {
+            return mIcons.get(position);
+        }
+
+        @Override
+        public int getCount() {
+            return mIcons.size();
+        }
+
+        @Override
+        public long getItemId(int position) {
+            return getItem(position).hashCode();
+        }
+
+        @NonNull
+        @Override
+        public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
+            final View view;
+            if (convertView == null) {
+                view = LayoutInflater.from(parent.getContext()).inflate(R.layout.custom_icon_item, parent, false);
+            } else {
+                view = convertView;
+            }
+            ViewHolder holder = view.getTag() instanceof ViewHolder ? (ViewHolder) view.getTag() : new ViewHolder(view);
+
+            IconData content = getItem(position);
+            holder.setContent(content);
+
+            holder.icon.setOnClickListener(v -> {
+                if (mOnItemClickListener != null)
+                    mOnItemClickListener.onItemClick(IconAdapter.this, v, position);
+            });
+            holder.icon.setOnLongClickListener(v -> {
+                displayToast(v, content.drawableInfo.getDrawableName());
+                return true;
+            });
+
+            return view;
+        }
+
+        /**
+         * @param v       is the Button view that you want the Toast to appear above
+         * @param message is the string for the message
+         */
+
+        private void displayToast(View v, CharSequence message) {
+            int xOffset = 0;
+            int yOffset = 0;
+            Rect gvr = new Rect();
+
+            //View parent = (View) v.getParent();
+            //int parentHeight = parent.getHeight();
```

Comment:
Commented code

Reply:
The `displayToast` method is not showing the toast at the right `y` value when the dialog is not filling the screen, I was still looking for a fix and didn't want to lose the original code